app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,20 @@
+class ApplicationsController < ApplicationController
+  def new
+  end
+
+  def create
+    application = Applicant.create(application_params)
+    if !application.save
+      flash[:errors] = application.errors.full_messages
+      return redirect_to "/applications/new"
+    end
+    flash[:notice] = "Application has gone through"
+    favorites.pet_data.clear
```

Comment:
I abstracted a method called remove_all into the model for the clear function. You could implement that here if you want to make it more MVC

Reply:
Let me change this. Thanks for pointing that to me.